QueueImplementation/Business/Helper/CollectionType.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace Business.Helper
+{
+    /// <summary>
+    /// Represents the types of possible collections.
+    /// </summary>
+    /// <owner>Anton Petrenko</owner>
+    public enum CollectionType
```

Comment:
As I remember, an array and a linked-lest types are basic structures of the Queues and Stack that are implemented.
So when you store all this stuff in the same enum you make it confusing.

Reply:
Added separate enum for custom collections.